user/views.py

```diff
@@ -1,13 +1,97 @@
-import json
-import jwt
-import bcrypt
+import json, re, bcrypt, jwt
+from json.decoder import JSONDecodeError
 
+from django.views            import View 
+from django.http             import JsonResponse
+from django.db               import transaction
+from django.core.exceptions  import ValidationError
+
+from user.models import User, Address
 from my_settings import SECRET_KEY
 
-from django.views import View
-from django.http  import JsonResponse
 
-from user.models import User
+# 문제점
+# 1. email과 같은 key 값이 2개 이상 들어올 때는??
+# 2. 이벤트 SNS, 메시지 수신 여부
+class SignUpView(View):
+    @transaction.atomic
+    def post(self, request):    
+        try:
+            data = json.loads(request.body)
+            
+            # 회원 개인 정보
+            email         = data["email"]
+            password      = data["password"]
+            name          = data["name"]
+            phone_number  = data["phone_number"]
+            date_of_birth = data.get("date_of_birth") # null True
+
+            # 주소 관련 정보
+            zip_code       = data["zip_code"]
+            address        = data["address"]
+            detail_address = data.get("detail_address") # null True
+
+            # 이메일 정규식 검사
+            regex_for_email = re.compile('^[a-zA-Z0-9+-_.]+@[a-zA-Z0-9-]+\.[a-zA-Z0-9-.]+$')
+
+            
+            signup_fail_response = JsonResponse({"message": "Bad Request"}, status=400)
+
+            if not email:
+                return signup_fail_response
+
+            if not regex_for_email.match(email):
+                return signup_fail_response
+            
+            if User.objects.filter(email=email).exists():
+                return JsonResponse({"message": "Conflict"}, status=409)
+            
+            # 비밀번호 검사
+            if not password:
+                return signup_fail_response
+
+            if len(password) < 10:
```

Comment:
```suggestion
            if len(password) < MIN_PASSWORD_LENGTH:
```
비교조건에 있는 항목이 하드코딩되지 않도록 임포트 열 하단에  상수로 선언해서 적용해주세요.

Reply:
네 알겠습니다~!